### PR TITLE
added facade users getter

### PR DESCRIPTION
**Pull request by Desmondmies**
- [x] Add Users Getter in Facade ( Student, Professor, Admin )
- [x] Modify corresponding DAOs, getting an Entity via email & password
- [x] Testing it all

### CODE REVIEW

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 1]
```diff
@@ -105,6 +110,60 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
...
 
+    /**
+     * Get Student of Database via email & password
+     * @param email Email of the student
+     * @param password password of the student
+     * @return Student entity if exist in database, otherwise return null.
+     */
+    public static Student getStudentFromAuth(String email, String password){
+        StudentDAO dao = StudentDAO.getInstance();
+        Student student = null;
```
Comment (anastygnome) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:121]: I'd prefer an Optional here, this is a perfect example of a good usecase.

**Review by anastygnome — CHANGES_REQUESTED** [review]
I'd like the getter to return an Optional rather than a null but other than that this is good !

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 1 above)
Comment (Desmondmies) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:121]: i thought about it while writing it, and for me from the facade Schedule it's unecessary, the dao alreay returns an optional which will throw an exception if null
but fine.

Comment (anastygnome) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java]: Let's just return that optional then, I believe the caller should handle the exception at that level.

**Review by anastygnome — APPROVED** [review]

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java (hunk 1 above)
Comment (Theo-Hafsaoui) [reply edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java]: Here, I agree with @anastygnome

File: edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java [hunk 2]
```diff
@@ -105,6 +111,60 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
...
+     * @param email Email of the student
+     * @param password password of the student
+     * @return Student entity if exist in database, otherwise return null.
+     */
+    public static Optional<Student> getStudentFromAuth(String email, String password){
+        StudentDAO dao = StudentDAO.getInstance();
+        Student student = null;
+
+        try {
+            student = dao.get(email, password).orElseThrow(SQLException::new);
```
Comment (Theo-Hafsaoui) [inline edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java:125]: I think in the future we should hash the pwd in the facade, but of course only when we tackle #47

**Review by Theo-Hafsaoui — APPROVED** [review]
gj

**Pull request merged**